File: java/leetcode/src/main/java/com/leetcode/BST/Kth_Smallest_Element_in_a_BST_230.java
package com.leetcode.BST;

public class Kth_Smallest_Element_in_a_BST_230 {
    boolean found = false;
    int target = 0;
    int _k = 0;
    int cnt = 0;
    public int kthSmallest(TreeNode root, int k) {
        _k = k;
        dfs(root);
        return target;
    }

    void dfs(TreeNode node) {
        if (node == null) return;
        dfs(node.left);
        cnt++;
        if (cnt == _k) {
            found = true;
            target = node.val;
        }
        if (found) return;
        dfs(node.right);
    }
}

File: js/algrithm/huahua130/BST/Kth_Smallest_Element_in_a_BST_230.js

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
    let index = 0
    let ans = undefined
    const dfs = function(node) {
        if (node === null) return
        dfs(node.left)
        index++
        if (index === k) {
            ans = node.val
            return
        }
        dfs(node.right)
    }

    dfs(root)

    return ans
    
};